File: UI/ems-ui/src/services/departmentService.js
import api from './api';

export const getAll = async () => {
    const response = await api.get('/departments');
    return response.data;
};

export const getById = async (id) => {
    const response = await api.get(`/departments/${id}`);
    return response.data;
};

export const create = async (deptData) => {
    const response = await api.post('/departments', deptData);
    return response.data;
};

export const update = async (id, deptData) => {
    const response = await api.put(`/departments/${id}`, deptData);
    return response.data;
};

export const remove = async (id) => {
    const response = await api.delete(`/departments/${id}`);
    return response.data;
};

File: API/ems/src/main/java/com/agile/ems/departments/service/DepartmentService.java
package com.agile.ems.departments.service;

import com.agile.ems.departments.dto.DepartmentRequestDto;
import com.agile.ems.departments.dto.DepartmentResponseDto;
import java.util.List;

public interface DepartmentService {
    DepartmentResponseDto create(DepartmentRequestDto requestDto);

    DepartmentResponseDto getById(Long id);

    List<DepartmentResponseDto> getAll();

    DepartmentResponseDto update(Long id, DepartmentRequestDto requestDto);

    void delete(Long id);
}
